apps/web/src/hooks/queries/use-search-movies-query.ts

```diff
@@ -1,10 +1,20 @@
-import { getSearchMovies } from '@/remotes/movie/get-search-movies';
 import { useQuery } from '@tanstack/react-query';
+import { AxiosResponse } from 'axios';
+
+import { SearchMoviesModel } from '@/model/search-movies';
+import { apiRoutesClient } from '@/remotes/common/api-routes-client';
+import { GetSearchMoviesResponse } from '@/remotes/movie/get-search-movies';
 
 export default function useSearchMoviesQuery(query: string, options?: { enabled?: boolean }) {
   return useQuery({
     queryKey: ['movies', query],
-    queryFn: () => getSearchMovies({ query, page: 1 }),
+    queryFn: async () => {
+      const { data } = (await apiRoutesClient.get('/api/movie/get-search-movies', {
+        params: { query, page: 1 },
+      })) as AxiosResponse<GetSearchMoviesResponse>;
+
+      return data.results.map((movie) => new SearchMoviesModel(movie));
+    },
```

Comment:
타입 단언을 많이 사용하는 것으로 보이는데 최대한 지양해서 작성하면 좋겠습니다.

```suggestion
      const getSearchMovies = (): Promise<GetSearchMoviesResponse> => apiRoutesClient.get('/api/movie/get-search-movies', {
        params: { query, page: 1 },
      }).then((res) => res.data)
      
      const { results } = await getSearchMovies()

      return results.map((movie) => new SearchMoviesModel(movie));
    },
```